cores/arduino/api/Print.h

```diff
@@ -95,18 +83,14 @@ class Print {
     size_t println(unsigned int, int = DEC);
     size_t println(long, int = DEC);
     size_t println(unsigned long, int = DEC);
+    size_t println(long long, int = DEC);
+    size_t println(unsigned long long, int = DEC);
     size_t println(double, int = 2);
-    size_t println(const Printable &);
+    size_t println(const Printable&);
     size_t println(void);
-#ifdef SUPPORT_LONGLONG
-    void println(int64_t, uint8_t = DEC);
-    void print(int64_t, uint8_t = DEC);
-    void println(uint64_t, uint8_t = DEC);
-    void print(uint64_t, uint8_t = DEC);
-#endif
-
-    int printf(const char *format, ...);
-    int printf(const __FlashStringHelper *format, ...);
```

Comment:
`printf()` support was deleted? Should be added back in later commit.

Reply:
I'm happy for us to add it back, but `printf` to the best of my knowledge, isn't actually part of the Arduino core standard.

Reply:
Right, AVR definitely wouldn't like that with that code size, but here it should be permissible, and users from STM32Duino and other cores (ESP32, ..) might expect it to be there since it was there, too. If it has meaningful flash impact we might wrap it into a configurable `#define/#ifdef GD32_ENABLE_SERIAL_PRINTF`.

Reply:
It is indeed desirable. We should figure out how Arduino wants folks to do it in the new Arduino-API world